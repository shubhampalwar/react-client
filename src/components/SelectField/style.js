

const style = {
  base: {
    height: '20px',
    padding: '20px',
    borderRadius: '5px',
    border: '1px solid lightgrey',
  },
  error: {
    border: '1px solid red',
  },
  option: {
    height: '20px',
    padding: '20px',
    borderRadius: '5px',
    border: '1px solid lightgrey',
  },
};

export default style;
