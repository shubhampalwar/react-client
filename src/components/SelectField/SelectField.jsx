import React, { Component } from 'react';
import PropTypes from 'prop-types';
import style from './style';

class SelectField extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const {
      error, options, defaultText, title, ...rest
    } = this.props;
    const errorStyle = error ? { ...style.error } : {};
    return (
      <>
        <div style={style.title}>{title}</div>
        <select
          value={defaultText}
          {...rest}
          style={{ ...style.base, ...errorStyle }}
        >
          <option style={style.base} value="" disabled>
            {defaultText}
          </option>
          {options.map(option => (
            <option key={option.value} value={option.value}>
              {option.label}
            </option>
          ))}
        </select>
        <div style={style.errorText}>{error}</div>
      </>
    );
  }
}

SelectField.propTypes = {
  error: PropTypes.string,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({ label: PropTypes.string, value: PropTypes.string }),
  ),
  defaultText: PropTypes.string,
  title: PropTypes.string.isRequired,
};
SelectField.defaultProps = {
  defaultText: 'Select',
  error: '',
  options: [],
};
export default SelectField;
