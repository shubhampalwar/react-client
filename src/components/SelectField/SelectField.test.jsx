import { shallow } from 'enzyme';
import React from 'react';
import SelectField from './SelectField';

const wrapper = shallow(<SelectField />);

describe('Select Field', () => {
  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
