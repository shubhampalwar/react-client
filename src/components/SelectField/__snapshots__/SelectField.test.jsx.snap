// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Field Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectField
    defaultText="Select"
    error=""
    options={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "fontSize": "21px",
              "fontWeight": "bold",
            }
          }
        />,
        <select
          style={
            Object {
              "border": "1px solid lightgrey",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "margin": "5px 0px 0px 0px",
              "padding": "12px",
              "width": "100%",
            }
          }
          value="Select"
        >
          <option
            disabled={true}
            style={
              Object {
                "border": "1px solid lightgrey",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "margin": "5px 0px 0px 0px",
                "padding": "12px",
                "width": "100%",
              }
            }
            value=""
          >
            Select
          </option>
        </select>,
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "color": "red",
              "padding": "10px",
            }
          }
        >
          
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "style": Object {
            "fontSize": "21px",
            "fontWeight": "bold",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              style={
                Object {
                  "border": "1px solid lightgrey",
                  "borderRadius": "5px",
                  "boxSizing": "border-box",
                  "margin": "5px 0px 0px 0px",
                  "padding": "12px",
                  "width": "100%",
                }
              }
              value=""
            >
              Select
            </option>,
            Array [],
          ],
          "style": Object {
            "border": "1px solid lightgrey",
            "borderRadius": "5px",
            "boxSizing": "border-box",
            "margin": "5px 0px 0px 0px",
            "padding": "12px",
            "width": "100%",
          },
          "value": "Select",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select",
              "disabled": true,
              "style": Object {
                "border": "1px solid lightgrey",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "margin": "5px 0px 0px 0px",
                "padding": "12px",
                "width": "100%",
              },
              "value": "",
            },
            "ref": null,
            "rendered": "Select",
            "type": "option",
          },
        ],
        "type": "select",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "style": Object {
            "boxSizing": "border-box",
            "color": "red",
            "padding": "10px",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "fontSize": "21px",
                "fontWeight": "bold",
              }
            }
          />,
          <select
            style={
              Object {
                "border": "1px solid lightgrey",
                "borderRadius": "5px",
                "boxSizing": "border-box",
                "margin": "5px 0px 0px 0px",
                "padding": "12px",
                "width": "100%",
              }
            }
            value="Select"
          >
            <option
              disabled={true}
              style={
                Object {
                  "border": "1px solid lightgrey",
                  "borderRadius": "5px",
                  "boxSizing": "border-box",
                  "margin": "5px 0px 0px 0px",
                  "padding": "12px",
                  "width": "100%",
                }
              }
              value=""
            >
              Select
            </option>
          </select>,
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "color": "red",
                "padding": "10px",
              }
            }
          >
            
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "style": Object {
              "fontSize": "21px",
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                disabled={true}
                style={
                  Object {
                    "border": "1px solid lightgrey",
                    "borderRadius": "5px",
                    "boxSizing": "border-box",
                    "margin": "5px 0px 0px 0px",
                    "padding": "12px",
                    "width": "100%",
                  }
                }
                value=""
              >
                Select
              </option>,
              Array [],
            ],
            "style": Object {
              "border": "1px solid lightgrey",
              "borderRadius": "5px",
              "boxSizing": "border-box",
              "margin": "5px 0px 0px 0px",
              "padding": "12px",
              "width": "100%",
            },
            "value": "Select",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select",
                "disabled": true,
                "style": Object {
                  "border": "1px solid lightgrey",
                  "borderRadius": "5px",
                  "boxSizing": "border-box",
                  "margin": "5px 0px 0px 0px",
                  "padding": "12px",
                  "width": "100%",
                },
                "value": "",
              },
              "ref": null,
              "rendered": "Select",
              "type": "option",
            },
          ],
          "type": "select",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "style": Object {
              "boxSizing": "border-box",
              "color": "red",
              "padding": "10px",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
